src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -194,6 +194,13 @@ void CreatePrimaryRequest<I>::unlink_peer() {
       if (info == nullptr) {
         continue;
       }
+      if (info->state == cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY_DEMOTED &&
+          m_image_ctx->snap_info.size() > 1 &&
```

Comment:
I don't think checking for `m_image_ctx->snap_info.size() > 1` is needed.  `UnlinkPeerRequest` already does that:
https://github.com/ceph/ceph/blob/704363b4a18fc9d0925dc2dbc93f20de564eb988/src/librbd/mirror/snapshot/UnlinkPeerRequest.cc#L100-L106
`remove_snapshot()` can be called only if `have_newer_mirror_snapshot` is set, so we don't need to be strict in `CreatePrimaryRequest`.